assets/src/edit-story/components/canvas/canvasProvider.js

```diff
@@ -20,26 +33,67 @@ function CanvasProvider( { children } ) {
 		setNodeForElement,
 	} = useEditingElement();
 
-	const [
-		backgroundMouseDownHandler,
-		setBackgroundMouseDownHandler,
-		clearBackgroundMouseDownHandler,
-	] = useFunctionState();
+	const {
+		state: { currentPage, selectedElementIds },
+		actions: { selectElementById, toggleElementIdInSelection, setSelectedElementIds },
+	} = useStory();
+
+	const handleSelectElement = useCallback( ( elId, evt ) => {
+		if ( editingElement && editingElement !== elId ) {
+			clearEditing();
+		}
+
+		if ( evt.metaKey ) {
+			toggleElementIdInSelection( elId );
+		} else {
+			selectElementById( elId );
+		}
+		evt.stopPropagation();
+
+		if ( 'mousedown' === evt.type ) {
+			evt.persist();
+			setLastSelectionEvent( evt );
+		}
+	}, [ editingElement, clearEditing, toggleElementIdInSelection, selectElementById ] );
+
+	const selectIntersection = useCallback( ( { x: lx, y: ly, width: lw, height: lh } ) => {
+		const newSelectedElementIds =
+			currentPage.elements.filter( ( { x, y, width, height } ) => {
+				return (
+					x <= lx + lw &&
+					lx <= x + width &&
+					y <= ly + lh &&
+					ly <= y + height
+				);
+			} ).map( ( { id } ) => id );
+		setSelectedElementIds( newSelectedElementIds );
+	}, [ currentPage, setSelectedElementIds ] );
+
+	// Reset editing mode when selection changes.
+	useEffect( () => {
+		if ( editingElement &&
```

Comment:
In which case would it be true? Meaning, in which case would an element be in edit mode and multiple elements being selected at the same time?

Reply:
The selection and editing are managed on different levels: the selection is managed by the reducer model in the `StoryProvider` and the edit mode is managed in the `Canvas` below it. I thought it was very likely that a selection could change via story model action (`setSelectedElementIds` method), while edit element remains unchanged. In reality, however, edit element must always also be selected one and it makes no sense to have an edit element for multiple selections. So this effect is a basically catch-all way to exit edit mode when the selection changes.